### PR TITLE
Run python tests during make check

### PR DESCRIPTION
**Pull request by makaimann**
If building with python bindings, check the `pytest` is installed, and adds a command to run `pytest` after the existing `make check` tests. If built without python bindings, it just uses a null command. Note: the current semantics are such that the `pytest` tests will not run if the `ctest` run fails (unless you pass the correct flag to make to continue `--ignore-errors` I believe). I can look into changing this semantics if that would be preferred.

### CODE REVIEW

**Review by aniemetz — APPROVED** [review]
LGTM. Makes sense that it doesn't run if ctest fails -- this would have to be fixed first anyways.

**Review by mpreiner — CHANGES_REQUESTED** [review]
As discussed offline with @makaimann. The Python unit tests should be integrated into ctest (each unit test is one ctest test) as otherwise executing ctest does not execute Python unit tests.

**Review by aniemetz — CHANGES_REQUESTED** [review]
I just realized that this should be included in ctest rather than being called outside of ctest. @mpreiner just told me that you talked offline about this.

**Comment by makaimann** [comment]
All right, I updated this to run the tests through `ctest`. It runs all the python tests as one test - part of the `pytest` infrastructure is to automatically detect tests in a directory. We could separate them into multiple tests by listing the test files individually, but I'd vote not to do that because then it needs to be maintained.

**Comment by mpreiner** [comment]
An advantage of splitting up the tests is that they will be run in parallel with `ctest -j ...`. Further, if a Python test fails `ctest` will list the specific test case that failed and not `all Python tests failed`. Adding test files manually is not an issue in my opinion -  we already do that for all the other unit tests, it shouldn't be more than adding a new line to a CMake list.

**Comment by makaimann** [comment]
Thanks for the feedback -- that makes sense! I just update it so that tests are added individually.

**Review by mpreiner — CHANGES_REQUESTED** [review]
I think we should move the Python API tests to test/api/python instead of test/unit/api. One of the reasons is that we do not run unit tests if it is not a debug build. However, we can run the Python tests even in a production build.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ jobs:
     - name: ccache Statistics
       run: ccache -s
 
-    - name: Run CTest
+    - name: Run CTest (and Pytest if python bindings built)
```
[the remark this answers is not in the document]
Comment (makaimann) [reply .github/workflows/ci.yml]: Cool, removed it in 2f28e9cd60a245d85d8c9efdcea0b1860a49b797

**Comment by makaimann** [comment]
That's a good point about where to put the tests -- moved them in 37b7e540220a94ca0295694676b9d1b2271d526d

**Review by mpreiner — APPROVED** [review]

**Comment by mpreiner** [comment]
Thanks for all the changes!

**Review by aniemetz — APPROVED** [review]

**Pull request merged**